src/shared/utils/windowPeriod/fromAST.test.ts

```diff
@@ -0,0 +1,285 @@
+import {
+  getWindowPeriodVariableAssignment,
+  getWindowPeriodFromAST,
+} from './fromAST'
+import {getRangeVariable} from 'src/variables/utils/getTimeRangeVars'
+import {Package} from 'src/types/ast'
+import {TimeRange} from 'src/types'
+import {asAssignment} from 'src/variables/selectors'
+import {buildVarsOption} from 'src/variables/utils/buildVarsOption'
+import {TIME_RANGE_START, TIME_RANGE_STOP} from 'src/variables/constants'
+import {FALLBACK_WINDOW_PERIOD} from 'src/shared/constants/timeRanges'
+
+import textFixtures from './ast_fixtures.json'
+
+describe('Getting a windowPeriod from a flux query AST', () => {
```

Comment:
Test cases used in this file are an exact copy from the scoping issue ticket:
https://github.com/influxdata/ui/issues/4836